src/helpers/misc.ts

```diff
@@ -357,11 +357,13 @@ export function deepEquals(o1: any, o2: any): boolean {
   if (typeof o1 !== "object") return o1 === o2;
 
   // Objects can have different keys if the values are undefined
-  const keys = new Set<string>();
-  Object.keys(o1).forEach((key) => keys.add(key));
-  Object.keys(o2).forEach((key) => keys.add(key));
+  for (const key in o2) {
+    if (!(key in o1) && o2[key] !== undefined) {
```

Comment:
isn't this testing that keys in `o2` that are defined must be in `o1` ?
what about keys in `o1` that might not be in `o2`?

Reply:
> isn't this testing that keys in o2 that are defined must be in o1 ?

keys in `o2` that are not in `o1` must have their value `undefined` (otherwise it means the two objects are not equals)

> what about keys in o1 that might not be in o2?

In the for loop below, we iterate over keys of `o1` and check `if (o1[key] !== o2[key]) return false;`